### ISSUE TITLE
🐛 [flutterfire_ui] Email Verification feature missing in Email and Password authentication flow.

### ISSUE THREAD

**Issue by prajwal27** [firebase/flutterfire#8788]
## Bug report

**Describe the bug**
There's no feature to verify the email address from the built-in SignInScreen and RegisterScreen provided by the package. Is this work in progress or am I mistaken?

**Comment by darshankawar** [comment]
This was added as part of this PR https://github.com/firebase/flutterfire/pull/8358, so the feature must be there in the plugin already.

Closing for now. If you disagree, write in comments and I'll reopen it.

**Comment by prajwal27** [comment]
@darshankawar  the feature is present as a part of the Profile screen and not as part of user registration process.

Imagine a scenario where I would want the email to be verified before creating the firebase user with that email id. I don't think that's possible now.
I want to eradicate usage of dummy/spam email ids

**Comment by darshankawar** [comment]
Thanks for the feedback. Reopening.

**Comment by darshankawar** [comment]
Treating this as an enhancement. 

/cc @lesnitsky

**Comment by lesnitsky** [comment]
> Imagine a scenario where I would want the email to be verified before creating the firebase user with that email id. I don't think that's possible now.

I don't think this is possible at all given the underlying firebase auth API. @prajwal27 could you please provide an example of how would you implement such behaviour without flutterfire_ui?

**Comment by pabloalencar** [comment]
Looking forward to this feature. My concern in implementing email verification is to ensure that a user does not sign up with someone else's e-mail.

I would also like the functionality not to be implemented on the profile screen, as developers can make apps without a profile screen but still have email verification.

**Comment by prajwal27** [comment]
> The thing here is that when users sign up using the already provided EmailProviderConfiguration in FlutterFire RegisterScreen, a user account is successfully created which can be seen at the Authentication>>Users area in Firebase dashboard.
> 
> However, there's no way to confirm if the email a user signed up with is valid or not. An Email Verification link would solve this.
> 
> You could however create a solution around this. After users create an account, you could redirect them to a LandingScreen that would check whether to automatically send email verification using; "await user?.sendEmailVerification();".
> 
> A condition could be used to check if "user.emailVerified" is true or false before redirecting to specific screen.
> 
> If false, you could provide a link to resend email verification link. But if true, users would be redirected to the HomeScreen.

This is a great way to ensure that users verify their email before proceeding to the actual app. But what I'm concerned is about dummy/spam email ids being used at the time of registration. There will be lots of Firebase Users created from these emails and there's no way to prevent it.  @darshankawar  @lesnitsky

**Comment by mikejones3** [comment]
Waiting on this feature as well, I do not want users to "steal" other peoples emails addresses, preventing email address owners fromusign their own email address.

**Comment by francescovgg** [comment]
I do also believe emailVerification should be considered as a basic flow for security. Someone could have signed up with my email and this will prevent me to signup as well.

Looking forward for this.

**Comment by lesnitsky** [comment]
fixed in #9343
